subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1511,13 +1511,14 @@ void ull_adv_done(struct node_rx_event_done *done)
 		return;
 	}
 
+	handle = ull_adv_handle_get(adv);
+	LL_ASSERT(handle < BT_CTLR_ADV_SET);
+
 	rx_hdr->type = NODE_RX_TYPE_EXT_ADV_TERMINATE;
+	rx_hdr->handle = handle;
 	rx_hdr->rx_ftr.param_adv_term.conn_handle = 0xffff;
 	rx_hdr->rx_ftr.param_adv_term.num_events = adv->event_counter;
 
-	handle = ull_adv_handle_get(adv);
-	LL_ASSERT(handle < BT_CTLR_ADV_SET);
-
```

Comment:
These changes seem to be unrelated to scanning. I would prefer to move them to a separate PR (which then probably will be trivial to review and get merged).